tests/unit/plugins/module_utils/crypto/test_cryptography_support.py

```diff
@@ -31,3 +41,64 @@ def test_cryptography_get_name_other_name_utfstring():
     actual = cryptography_get_name('otherName:1.3.6.1.4.1.311.20.2.3;UTF8:Hello World')
     assert actual.type_id.dotted_string == '1.3.6.1.4.1.311.20.2.3'
     assert actual.value == b'\x0c\x0bHello World'
+
+
+@pytest.mark.parametrize('name, options, expected', [
+    (u'CN=x ', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u'x '), u'')),
+    (u'CN=\\ ', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u' '), u'')),
+    (u'CN=\\#', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u'#'), u'')),
+    (u'CN=#402032', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u'@ 2'), u'')),
+    (u'CN = x ', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u'x '), u'')),
+    (u'CN = x\\, ', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u'x, '), u'')),
+    (u'CN = x\\40 ', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u'x@ '), u'')),
+    (u'CN  =  \\  , / ', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u'  '), u', / ')),
+    (u'CN  =  \\  , / ', {'sep': '/'}, (NameAttribute(oid.NameOID.COMMON_NAME, u'  , '), u'/ ')),
+    (u'CN  =  \\  , / ', {'decode_remainder': False}, (NameAttribute(oid.NameOID.COMMON_NAME, u'\\  , / '), u'')),
+])
+def test_parse_dn_component(name, options, expected):
+    result = _parse_dn_component(name, **options)
+    print(result, expected)
+    assert result == expected
+
+
+# Cryptography < 2.9 does not allow empty strings
+# (https://github.com/pyca/cryptography/commit/87b2749c52e688c809f1861e55d958c64147493c)
+if LooseVersion(cryptography.__version__) >= LooseVersion('2.9'):
+    @pytest.mark.parametrize('name, options, expected', [
+        (u'CN=', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u''), u'')),
+        (u'CN= ', {}, (NameAttribute(oid.NameOID.COMMON_NAME, u''), u'')),
+    ])
+    def test_parse_dn_component_not_py26(name, options, expected):
+        result = _parse_dn_component(name, **options)
+        print(result, expected)
+        assert result == expected
+
+
+@pytest.mark.parametrize('name, options, message', [
+    (u'CN=\\0', {}, u'Hex escape sequence "\\0" incomplete at end of string'),
+    (u'CN=\\0,', {}, u'Hex escape sequence "\\0," has invalid second letter'),
+    (u'CN=#0,', {}, u'Invalid hex sequence entry "0,"'),
+])
+def test_parse_dn_component_failure(name, options, message):
+    with pytest.raises(OpenSSLObjectError, match=u'^%s$' % re.escape(message)):
+        result = _parse_dn_component(name, **options)
+
+
+@pytest.mark.parametrize('name, expected', [
+    (u'CN=foo', [NameAttribute(oid.NameOID.COMMON_NAME, u'foo')]),
+    (u'CN=foo,CN=bar', [NameAttribute(oid.NameOID.COMMON_NAME, u'foo'), NameAttribute(oid.NameOID.COMMON_NAME, u'bar')]),
+    (u'CN  =  foo ,  CN  =  bar', [NameAttribute(oid.NameOID.COMMON_NAME, u'foo '), NameAttribute(oid.NameOID.COMMON_NAME, u'bar')]),
+])
```

Comment:
Everything looks good to me. I think one non-ascii test-case, as in the [RFC examples](https://datatracker.ietf.org/doc/html/rfc4514#section-4), would be nice (really just a sanity check that non-ascii UTF-8 characters are not escaped).

Reply:
That was a really good idea, since it pointed out one flaw in the existing code: working with unicode strings doesn't allow to escape unicode endpoints with 8-bit hex digits. I've rewritten it to use byte strings, and added some examples from the RFC as tests. Looks much better now IMO.

Reply:
Nice!